### PR TITLE
Some small build cleanups

### PR DESCRIPTION
**Pull request by bantonsson**

### CODE REVIEW

File: gradle/spotbugs.gradle [hunk 1]
```diff
@@ -5,46 +5,44 @@ spotbugs {
...
-}
-
-// disable spotbugs for all test tasks
+// configure spotbugs for Main tasks and disable it for all other
 afterEvaluate {
-  tasks.withType(spotbugsTest.class).configureEach {
+  tasks.withType(spotbugsMain.class).configureEach {
     def name = it.name
-    if (name.endsWith("Test")) {
+    if (name.endsWith("Main") || name.endsWith("Main_java8")) {
```
Comment (tylerbenson) [inline gradle/spotbugs.gradle:12]: What are some example tasks where this would be false?

**Review by tylerbenson — APPROVED** [review]

File: gradle/spotbugs.gradle (hunk 1 above)
Comment (bantonsson) [reply gradle/spotbugs.gradle:12]: `spotbugsTest`, `spotbugsLatestDepTest`, `spotbugsJmh`, et.c.

Comment (tylerbenson) [reply gradle/spotbugs.gradle:12]: Why wouldn't those be filtered out by `withType(spotbugsMain.class)`? (Maybe not if `spotbugsMain` extends `spotbugsTest`).

Comment (bantonsson) [reply gradle/spotbugs.gradle:12]: The class of all the spotbugs tasks is the same. It's just the most convenient way to get hold of the `class`.

Comment (tylerbenson) [reply gradle/spotbugs.gradle:12]: Looks like they're all of type `SpotBugsTask`.
https://github.com/spotbugs/spotbugs-gradle-plugin/blob/master/src/main/groovy/com/github/spotbugs/snom/SpotBugsTask.groovy

**Review by richardstartin — APPROVED** [review]

**Pull request merged**